### ISSUE TITLE
C# parse error on generic class 3.1 alpha 2

### ISSUE THREAD

**Issue by Jcozman** [godotengine/godot#23884]
**Godot version:**

3.1 alpha 2

**OS/device including version:**

Windows 10

**Issue description:**

Building a project with a generic class in C# fails because of a parse error. The message is the following:
```
 modules/mono/editor/csharp_project.cpp:170 - Parse error: Unexpected token: >
 Failed to determine namespace and class for script: res://Scripts/TestClass.cs
 modules/mono/editor/mono_bottom_panel.cpp:155 - Condition ' metadata_err != OK ' is true.
```
This is potentially related to https://github.com/godotengine/godot/issues/23497 but I figured I would call out the generic class issue specifically.

**Steps to reproduce:**
Implement a generic class like the following example, and add it to the .csproj so it will be included in the build.

```
public class TestClass<T>
{
    private T _data;

    public void SetData(T data)
    {
        _data = data;
    }
}
````

Then click "Build Project" from the "Mono" tab and observe the error. 

**Minimal reproduction project:**

**Comment by neikeq** [comment]
I think it's a duplicate of #23545, which should be fixed once #23515 is merged.

**Comment by Jcozman** [comment]
Thanks! Will close then 👍
